src/optimize.d

```diff
@@ -796,8 +796,23 @@ extern (C++) Expression Expression_optimize(Expression e, int result, bool keepL
                 return;
             }
             // If e2 *could* have been an integer, make it one.
-            if (e.e2.op == TOKfloat64 && (e.e2.toReal() == cast(sinteger_t)e.e2.toReal()))
-                e.e2 = new IntegerExp(e.loc, e.e2.toInteger(), Type.tint64);
+            if (e.e2.op == TOKfloat64)
+            {
+                version (all)
```

Comment:
Do you have a specific suggestion I can add?

In principle, this can only be safely removed once DMD is no longer compilable with DMD 2.068 or earlier, which probably won't be in a while.